### ISSUE TITLE
Errors falsely being raised when overriding an abstract class property

### ISSUE THREAD

**Issue by dob9601** [microsoft/pyright#2243]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
Pyright (as of a more recent version) now raises an error when overriding an abstract property.

**To Reproduce**
```py
import abc


class Foo(abc.ABC):
    @property
    @abc.abstractmethod
    def attribute(self) -> str:
        raise NotImplementedError


class Bar(Foo):
    attribute = 'foobar'
```
Raises:
```py
temp.py:12:17 - error: Expression of type "Literal['foobar']" cannot be assigned to declared type "property"
    "Literal['foobar']" is incompatible with "property" (reportGeneralTypeIssues)
```
**Expected behavior**
No errors

**VS Code extension or command-line**
command line

**Additional context**

**Comment by erictraut** [comment]
This is intended behavior. You are declaring a property in the base class and attempting to override it with an attribute, not a property. The semantics of a property and an attribute are similar but differ in important ways, and they are not interchangeable. 

The correct solution here is to override an abstract property with a property.
```python
class Bar(Foo):
    def attribute(self) -> str:
        return 'foobar'
```

**Comment by dob9601** [comment]
I'm basing my usage off of the stackoverflow answer here: https://stackoverflow.com/questions/2736255/abstract-attributes-in-python 
Wrapping a function around it feels a bit redundant, though I do see what you mean

**Comment by erictraut** [comment]
Yes, this is a gray area where the Python typing specs don't provide clarity. Some Python type checkers may be "looser" in this respect, but it's generally not type safe to override a property with an attribute or vice versa.
